token/js/src/actions/amountToUiAmount.ts

```diff
@@ -28,3 +31,189 @@ export async function amountToUiAmount(
     }
     return err;
 }
+
+const ONE_IN_BASIS_POINTS = 10000;
+const SECONDS_PER_YEAR = 60 * 60 * 24 * 365.24;
+
+/**
+ * Convert amount to UiAmount for a mint with interest bearing extension without simulating a transaction
+ * This implements the same logic as the CPI instruction available in /token/program-2022/src/extension/interest_bearing_mint/mod.rs
+ * 
+ * @param amount                   Amount of tokens to be converted
+ * @param decimals                 Number of decimals of the mint
+ * @param currentTimestamp         Current timestamp in seconds
+ * @param lastUpdateTimestamp      Last time the interest rate was updated in seconds
+ * @param initializationTimestamp  Time the interest bearing extension was initialized in seconds
+ * @param preUpdateAverageRate     Interest rate in basis points (1 basis point = 0.01%) before last update
+ * @param currentRate              Current interest rate in basis points
+ * 
+ * @return Amount scaled by accrued interest as a string with appropriate decimal places
+ */
+
+export function amountToUiAmountWithoutSimulation(
+    amount: string,
+    decimals: number,
+    currentTimestamp: number, // in seconds
+    lastUpdateTimestamp: number,
+    initializationTimestamp: number,
+    preUpdateAverageRate: number,
+    currentRate: number,
+): string {
+    Decimal.set({ toExpPos: 24, toExpNeg: -24 })
+
+    // Calculate pre-update exponent
+    const preUpdateTimespan = new Decimal(lastUpdateTimestamp).minus(initializationTimestamp);
+    const preUpdateNumerator = new Decimal(preUpdateAverageRate).times(preUpdateTimespan);
+    const preUpdateExponent = preUpdateNumerator.div(new Decimal(SECONDS_PER_YEAR).times(ONE_IN_BASIS_POINTS));
+    const preUpdateExp = preUpdateExponent.exp();
+
+    // Calculate post-update exponent
+    const postUpdateTimespan = new Decimal(currentTimestamp).minus(lastUpdateTimestamp);
+    const postUpdateNumerator = new Decimal(currentRate).times(postUpdateTimespan);
+    const postUpdateExponent = postUpdateNumerator.div(new Decimal(SECONDS_PER_YEAR).times(ONE_IN_BASIS_POINTS));
+    const postUpdateExp = postUpdateExponent.exp();
+
+    // Calculate total scale
+    const totalScale = preUpdateExp.times(postUpdateExp);
+
+    // Calculate scaled amount with interest rounded down to the nearest unit
+    const decimalsFactor = new Decimal(10).pow(decimals);
+    const scaledAmountWithInterest = new Decimal(amount).times(totalScale).div(decimalsFactor).toDecimalPlaces(decimals, Decimal.ROUND_DOWN);
+
+    return scaledAmountWithInterest.toString();
+}
+
+/**
+ * Convert amount to UiAmount for a mint with interest bearing extension without simulating a transaction
```

Comment:
nit: this works for any mint!
```suggestion
 * Convert amount to UiAmount for a mint without simulating a transaction
```